src/api/app/services/staging/staging_project_creator.rb

```diff
@@ -0,0 +1,47 @@
+module Staging
+  class StagingProjectCreator
+    def initialize(xml, staging_workflow)
+      @xml_hash = (Xmlhash.parse(xml) || {}).with_indifferent_access
+      @staging_workflow = staging_workflow
+    end
+
+    def create
+      staging_projects = []
+      staging_project_names.each do |name|
+        project = Project.find_or_initialize_by(name: name)
+        if project.staging
+          errors << "Project \"#{project}\": is a main project of a staging workflow."
```

Comment:
~~Project \"#{project}\": is a main project of a staging workflow.~~

Project \"#{project}\": is has a staging already. Nested stagings are not supported.